src/ray/gcs/redis_context.h

```diff
@@ -120,23 +121,31 @@ class RedisCallbackManager {
   struct CallbackItem : public std::enable_shared_from_this<CallbackItem> {
     CallbackItem() = default;
 
-    CallbackItem(const RedisCallback &callback, bool is_subscription, int64_t start_time,
+    CallbackItem(const RedisCallback &callback,
+                 std::optional<std::string> subscribe_channel, int64_t start_time,
                  instrumented_io_context &io_service)
         : callback_(callback),
-          is_subscription_(is_subscription),
+          subscribe_channel_(subscribe_channel),
           start_time_(start_time),
           io_service_(&io_service) {}
 
     void Dispatch(std::shared_ptr<CallbackReply> &reply) {
       std::shared_ptr<CallbackItem> self = shared_from_this();
       if (callback_ != nullptr) {
-        io_service_->post([self, reply]() { self->callback_(std::move(reply)); },
-                          "RedisCallbackManager.DispatchCallback");
+        if (subscribe_channel_) {
+          io_service_->post([self, reply]() { self->callback_(std::move(reply)); },
+                            "RedisCallbackManager.Pubsub." + *subscribe_channel_);
+        } else {
+          io_service_->post([self, reply]() { self->callback_(std::move(reply)); },
+                            "RedisCallbackManager.DispatchCallback");
+        }
```

Comment:
Are these changes related?

Reply:
Yes, if we'd like to cover redis pubsub callback. Do you think it's good to have it here? I can remove this since we are moving away from redis.